#Loops

In [ ]:
i= 1
while i<=10:     #print ujjwal 10 times
  print("Ujjwal")
  i = i+1
i =1
while i<=10:
  print(i,end = " ") #print no. 1 to 10 in a row
  i = i+1

Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
Ujjwal
1 2 3 4 5 6 7 8 9 10 

#Reverse Identation for reverse loop

In [ ]:
i= 10
while i>=1:
  print(i)
  i = i-1

10
9
8
7
6
5
4
3
2
1


Q. WAP to print even  numbers.

In [ ]:
i =1
while i<=20:     #print Even NO.
  if i % 2 == 0:
    print(i,end=" ")
  i = i+1

2 4 6 8 10 12 14 16 18 20 

#Sum of n numbers


In [ ]:
n = int(input("Enter a number for sum:"))
Sum = 0
i = 1            #i is starting point
while i <= n:       #condition
  Sum = Sum+i      #formula of sum
  i+=1             #Step to take
print(Sum)


Enter a number for sum:5
15


In [ ]:
n = int(input("Enter a number for sum:"))
Sum = 0
i = 1
while i <= n:
  Sum = Sum+i
  i+=1
print("Sum =", Sum)

Enter a number for sum:5
Sum = 15


#To find factorial


In [ ]:
#to find factorial
num = int(input("Enter a num to check factorial:"))

if num < 0:
  print("NEGATIVE NO.")
elif num == 0 or num == 1:       #fact of 1 and 0 is 1
  print("Factorial is 1")
else :
  factorial = 1
  for i in range(1,num+1):
    factorial =  factorial*i
    print(i," ",factorial)

Enter a num to check factorial:5
1   1
2   2
3   6
4   24
5   120


In [ ]:
#different factorial question
num = int(input("Enter a num to check factorial:"))
fact = 1
i = 1
while i<=num:
  fact = fact*i
  i+=1
  print(i," ",fact)

Q. WAP to check entered no. is positive or negative.

In [ ]:

num= 0
while num>=0:
  num = int(input("Enter a num:"))
print("Negative no")


Enter a num:6
Positive no.
Enter a num:6
Positive no.
Enter a num:-4
Positive no.
Negative no


In [ ]:
a= [2,4,6,8,10,12]        #print element of list according to index
idx = 0
while idx<len(a):
  print(idx,' ',a[idx])
  idx += 1

0   2
1   4
2   6
3   8
4   10
5   12


Q. WAP to find 4 exist in a list or not.

In [ ]:
n = [2,6,8,10,12,3,5,6,6,4,3,2,2]  #does 4 exists
idx = 0
while idx<len(n):
  if n[idx] == 4:
    print("Element found at index:",idx )
    break

  else:
    print("Element not found")
  idx+=1

Element not found
Element not found
Element not found
Element not found
Element not found
Element not found
Element not found
Element not found
Element not found
Element found at index: 9


In [ ]:
#question using break statament
n = [2,6,8,10,12,3,5,6,6,4,3,2,2]  #does 4 exists
idx = 0
while idx<len(n):
  if n[idx] == 4:
    print("Element found at index:",idx )
    break

  else:
    print("Element not found")
  idx+=1

In [ ]:
A = [1,2,3,4,5]      #diff type of output using iteration and list name A
for i in A:
  # print(i,end = " ")         # 1 2 3 4 5
  # print(A,end = " ")            # [1, 2, 3, 4, 5] [1, 2, 3, 4, 5] [1, 2, 3, 4, 5] [1, 2, 3, 4, 5] [1, 2, 3, 4, 5]
  print(A[i])

2
3
4
5


IndexError: list index out of range

Q. WAP to print table of digit n input by user upto 10.

In [ ]:
#table print
n = int(input("Enter a number:"))
for i in range(1,11,1):
  print(f"{n} x {i} = {n*i}")

Enter a number:2
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20


Q. WAP to print even no. from 2 to 20

In [ ]:
#even number
for i in range(2,21,2):
  print(f"{i}",end=" ")

2 4 6 8 10 12 14 16 18 20 

In [ ]:
n = 1
while n<=10:

#HW

Q1. get a digit from user eg-1,2,3,4 and reverse it without using reverse and slicing.

In [ ]:
n = input("Enter a number to check:")
#n = "1234"
reversed_n = ""
idx = len(n)

#print("Reverse number is :",end=" ")
#1.
while idx>0:
  reversed_n+= n[idx-1]
  idx -= 1
print("Reverse number is :",reversed_n)

#2.
if n == reversed_n:
  print("Number is palindrome")
else:
  print("Number is not palindrome")



Enter a number to check:1234
Reverse number is : 4321
Number is not palindrome


In [ ]:
n = int(input("Enter a number :"))
#n = 7
i = 1

while i <= n:
  print(f" Table for {i}: \t ",end="")
  i += 1

print()

j = 1
while j<= 10:
  i = 1

  while i<=n:
    print(f"{i} x {j} = {i*j}\t",end="")

    i +=1
  print()
  j += 1



Enter a number :5
 Table for 1: 	  Table for 2: 	  Table for 3: 	  Table for 4: 	  Table for 5: 	 
1 x 1 = 1	2 x 1 = 2	3 x 1 = 3	4 x 1 = 4	5 x 1 = 5	
1 x 2 = 2	2 x 2 = 4	3 x 2 = 6	4 x 2 = 8	5 x 2 = 10	
1 x 3 = 3	2 x 3 = 6	3 x 3 = 9	4 x 3 = 12	5 x 3 = 15	
1 x 4 = 4	2 x 4 = 8	3 x 4 = 12	4 x 4 = 16	5 x 4 = 20	
1 x 5 = 5	2 x 5 = 10	3 x 5 = 15	4 x 5 = 20	5 x 5 = 25	
1 x 6 = 6	2 x 6 = 12	3 x 6 = 18	4 x 6 = 24	5 x 6 = 30	
1 x 7 = 7	2 x 7 = 14	3 x 7 = 21	4 x 7 = 28	5 x 7 = 35	
1 x 8 = 8	2 x 8 = 16	3 x 8 = 24	4 x 8 = 32	5 x 8 = 40	
1 x 9 = 9	2 x 9 = 18	3 x 9 = 27	4 x 9 = 36	5 x 9 = 45	
1 x 10 = 10	2 x 10 = 20	3 x 10 = 30	4 x 10 = 40	5 x 10 = 50	


# HW 1234 is input  \
1. print 4321    \
2. is pallindrome   \
3. 4+3+2+1     \
4. string is pallindrome

In [ ]:
n = int(input("Enter a number:"))
ori_num = n
reverse = 0
digit_sum = 0
reverse_str = ""
str_num = str(n)
ori_str = str_num
while n>0:

  digit = n%10
  reverse = reverse * 10 + digit
  n = n //10
print(n)
print("reverse:",reverse)

if reverse == ori_num:
  print("is pallindrome")
else:
  print("not pallindrome")

  digit_sum = digit_sum + digit
  print("SUM:",digit_sum)

for char in str_num:
  reverse_str = char + reverse_str
print("Reversed str:",reverse_str)

if ori_str == reverse_str:
  print("is pallindrome")
else:
  print("not pallindrome")





Enter a number:12321
0
reverse: 12321
is pallindrome
Reversed str: 12321
is pallindrome


In [ ]:
#same last ques but with different approach

n = int(input("Enter a number:"))
ori_num = n
reverse = 0
digit_sum = 0

while n>0:

  digit = n%10
  reverse = reverse * 10 + digit
  n = n //10
print("reverse:",reverse)

if reverse == ori_num:
  print("is pallindrome")
else:
  print("not pallindrome")









Enter a number:456
reverse: 654
not pallindrome


#HW Practice Questions

1. Armstrong Number in a Range.  .
Print all Armstrong numbers between 1 and N.

In [ ]:
#Armstrong number

print("Armstrong numbers between 1 and 500:")
num = 0
while num <=500:
  num+=1
  str_num = str(num)
  n = len(str_num)

  arm_num = sum(int(digit)**n for digit in str_num)
  if arm_num == num:
    print(num,arm_num)


Armstrong numbers between 1 and 500:
1 1
2 2
3 3
4 4
5 5
6 6
7 7
8 8
9 9
153 153
370 370
371 371
407 407


2. Happy Number \
A number is called Happy if repeatedly replacing it with the sum of the squares of its digits eventually reaches 1.

In [ ]:
num = int(input("Enter a numnber to start happy number loop: "))
ori_num = num

while num != 1 and num!= 4:
  num = sum(int(digit)**2 for digit in str(num))
  if num == 1:
    break

if num == 1 :
  print(f"{ori_num} : is a Happy Number")
else:
  print(f"{ori_num} : is a not Happy Number")


Enter a numnber to start happy number loop: 89
89 : is a not Happy Number
4


3. Find the Largest Digit in a Number

In [ ]:
while True:
  n = int(input("Enter a num:"))
  l = []

  for digit in str(n):
    l.append(int(digit))
  print(f"{max(l)} is the Largest digit")
break


Enter a num:8787
8 is the Largest digit


4. Password Strength Checker (Using Loops) \
Check whether a password contains:\
 Uppercase\
 Lowercase\
 Digit\
 Special Character\
 Minimum 8 characters

In [ ]:
password = input("Enter a password: ")

has_upper = False
has_lower = False
has_digit = False
has_special = False

has_valid_length = len(password) >= 8

for char in password:
  if char.isupper():
    has_upper = True
  elif char.islower():
    has_lower = True
  elif char.isdigit():
    has_digit = True
  elif not char.isalnum():
    has_special = True

if  has_valid_length and has_upper  and has_lower and has_digit  and has_special  :
  print(f"{password} : is a Strong password")
else:
  print("Weak password : Missing requirements")
  if not has_valid_length : print("- Must be at least 8 characters long")
  if not has_upper :         print("- Missing Uppercase")
  if not has_lower  :        print("- Missing Lowercase")
  if not has_digit   :       print("- Missing digit")
  if not has_special  :      print("- Missing special char.")




Enter a password: f4
Weak password : Missing requirements
- Must be at least 8 characters long
-Missing Uppercase
-Missing special char.


6. Print Pascal's Triangle\
 Print the first N rows of Pascal's Triangle.\
 Example (N = 5):

In [ ]:
n = 5
for i in range(n):
    val = 1
    for j in range(i + 1):
        print(val, end=" ")
        # Condition to calculate the next number in the same row
        val = val * (i - j) // (j + 1)
    print()  # New line for the next row


1 
1 1 
1 2 1 
1 3 3 1 
1 4 6 4 1 
